tests/native_api_tests/test_api_model_update.cpp

```diff
@@ -244,4 +253,638 @@ TEST_CASE_TEMPLATE(
+
+        result.add_buffer("asym_load", std::ssize(asym_load_id), std::ssize(asym_load_id), nullptr, nullptr);
+        result.add_attribute_buffer("asym_load", "id", asym_load_id.data());
+        result.add_attribute_buffer("asym_load", "node", asym_load_node.data());
+        result.add_attribute_buffer("asym_load", "status", asym_load_status.data());
+        result.add_attribute_buffer("asym_load", "type", asym_load_type.data());
+        result.add_attribute_buffer("asym_load", "p_specified", asym_load_p_specified.data());
+        result.add_attribute_buffer("asym_load", "q_specified", asym_load_q_specified.data());
+
+        result.add_buffer("shunt", std::ssize(shunt_id), std::ssize(shunt_id), nullptr, nullptr);
+        result.add_attribute_buffer("shunt", "id", shunt_id.data());
+        result.add_attribute_buffer("shunt", "node", shunt_node.data());
+        result.add_attribute_buffer("shunt", "status", shunt_status.data());
+        result.add_attribute_buffer("shunt", "g1", shunt_g1.data());
+        result.add_attribute_buffer("shunt", "b1", shunt_b1.data());
+        result.add_attribute_buffer("shunt", "g0", shunt_g0.data());
+        result.add_attribute_buffer("shunt", "b0", shunt_b0.data());
+
+        return result;
+    }
+};
+} // namespace
+
+TEST_CASE("API model - all updates") {
+    using namespace std::string_literals;
+
+    State state;
+    auto const input_dataset = state.get_input_dataset();
+    auto const& input_info = input_dataset.get_info();
+    auto model = Model{50.0, input_dataset};
+
+    // update vector
+    std::vector<ID> sym_load_update_id{7};
+    std::vector<IntS> sym_load_update_status{1};
+    std::vector<double> sym_load_update_p_specified{2.5e6};
+
+    std::vector<ID> asym_load_update_id{8};
+    std::vector<IntS> asym_load_update_status{0};
+
+    std::vector<ID> shunt_update_id{9};
+    std::vector<IntS> shunt_update_status{0};
+    std::vector<double> shunt_update_b1{0.02};
+    std::vector<double> shunt_update_b0{0.02};
+
+    // used for test case alternate compute mode
+    std::vector<ID> shunt_update_2_id{6};
+    std::vector<IntS> source_update_2_status{0};
+    std::vector<double> shunt_update_2_b1{0.01};
+    std::vector<double> shunt_update_2_b0{0.01};
+
+    std::vector<ID> source_update_id{10};
+    std::vector<IntS> source_update_status{1};
+    std::vector<double> source_update_u_ref{test::u1};
+
+    std::vector<ID> link_update_id{5};
+    std::vector<IntS> link_update_from_status{1};
+    std::vector<IntS> link_update_to_status{0};
+
+    DatasetConst update_data{"update", true, 1};
+    update_data.add_buffer("sym_load", 1, 1, nullptr, nullptr);
+    update_data.add_attribute_buffer("sym_load", "id", sym_load_update_id.data());
+    update_data.add_attribute_buffer("sym_load", "status", sym_load_update_status.data());
+    update_data.add_attribute_buffer("sym_load", "p_specified", sym_load_update_p_specified.data());
```

Comment:
Just curious: Any reason why everything is now columnar? I personally find it cleaner and more intuitive, but what is your reason? Is it to take advantage of the better memory usage?

Reply:
less lines + more intuitive for the time being. the alternatives are adding a lot of lines to set all attributes specifically (or providing overloads for them) but still with the same amount of lines.

Reply:
Definetly like it better myself. A lot more practical for us at least. Resolving.